bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/MapUtils.java

```diff
@@ -148,9 +151,53 @@ public static void checkIsMapOnlyOperation(BType mapType, String op) {
         switch (mapType.getTag()) {
             case TypeTags.MAP_TAG:
             case TypeTags.JSON_TAG:
+            case TypeTags.RECORD_TYPE_TAG:
                 return;
             default:
                 throw createOpNotSupportedError(mapType, op);
         }
     }
+
+    public static void checkValidFieldForRecord(MapValue<?, ?> m, String  k, String  op) {
+        BType type = m.getType();
+        switch (type.getTag()) {
+            case TypeTags.RECORD_TYPE_TAG:
+                if (!m.containsKey(k)) {
+                    return;
+                }
+                if (isRequiredField((BRecordType) type, k)) {
+                    throw createOpNotSupportedErrorForRecord(type, op);
+                }
+                return;
+            default:
+                return;
+        }
+    }
+
+    private static boolean isRequiredField(BRecordType type, String k) {
+        Map<String, BField> fields = type.getFields();
+        BField field = fields.get(k);
+        if (field != null && Flags.isFlagOn(field.flags, Flags.REQUIRED)) {
+            return true;
+        }
+        return false;
+    }
+
+    public static void validateRecord(BType type, String op) {
```

Comment:
This method is doing the same thing as the `checkValidFieldForRecord()` method right? Shall we get rid of one of these? Also simplify the logic as well

Reply:
This method validates on all the fields of the RecordType for RemoveAll implementation